src/code/z_message.c

```diff
@@ -223,339 +808,4230 @@ void Message_FindMessage(PlayState* play, u16 textId) {
+    msgCtx->textDelayTimer = 0;
+    msgCtx->textboxColorAlphaCurrent = msgCtx->textboxColorAlphaTarget = msgCtx->textColorAlpha = 0;
+    msgCtx->stateTimer = 30;
+
+    // Day/Dawn/Night.. Messages
+    if ((msgCtx->currentTextId >= 0x1BB2) && (msgCtx->currentTextId <= 0x1BB6)) {
+        XREG(74) = 0x6A;
+        XREG(75) = 0;
+        XREG(77) = 0x58;
+        XREG(76) = 0x44;
+    } else {
+        XREG(74) = 0x42;
+        XREG(75) = 0x1E;
+        XREG(77) = 0x3C;
+        XREG(76) = 0x1C;
+        msgCtx->unk11F1A[0] = msgCtx->unk11F1A[1] = msgCtx->unk11F1A[2] = 0;
+        Interface_SetHudVisibility(HUD_VISIBILITY_NONE);
+    }
+}
+
+void Message_BombersNotebookQueueEvent(PlayState* play, u8 event) {
+    MessageContext* msgCtx = &play->msgCtx;
+
+    if (CHECK_QUEST_ITEM(QUEST_BOMBERS_NOTEBOOK)) {
+        if (!CHECK_WEEKEVENTREG(gBombersNotebookWeekEventFlags[event])) {
+            msgCtx->bombersNotebookEventQueue[msgCtx->bombersNotebookEventQueueCount] = event;
+            msgCtx->bombersNotebookEventQueueCount++;
+        }
+    } else if (event >= BOMBERS_NOTEBOOK_PERSON_MAX) {
+        // Non MET events are processed even if the player does not have the notebook yet
+        if (!CHECK_WEEKEVENTREG(gBombersNotebookWeekEventFlags[event])) {
+            msgCtx->bombersNotebookEventQueue[msgCtx->bombersNotebookEventQueueCount] = event;
+            msgCtx->bombersNotebookEventQueueCount++;
+        }
+    }
+}
+
+/**
+ * @return s32 true if an entry to be shown was found and triggered
+ */
+s32 Message_BombersNotebookProcessEventQueue(PlayState* play) {
+    MessageContext* msgCtx = &play->msgCtx;
+
+    while (true) {
+        if (msgCtx->bombersNotebookEventQueueCount == 0) {
+            return false;
+        }
+        msgCtx->bombersNotebookEventQueueCount--;
+
+        if (!CHECK_WEEKEVENTREG(gBombersNotebookWeekEventFlags
+                                    [msgCtx->bombersNotebookEventQueue[msgCtx->bombersNotebookEventQueueCount]])) {
+            SET_WEEKEVENTREG(gBombersNotebookWeekEventFlags
+                                 [msgCtx->bombersNotebookEventQueue[msgCtx->bombersNotebookEventQueueCount]]);
+
+            if ((sBombersNotebookEventMessages
+                     [msgCtx->bombersNotebookEventQueue[msgCtx->bombersNotebookEventQueueCount]] != 0) &&
+                CHECK_QUEST_ITEM(QUEST_BOMBERS_NOTEBOOK)) {
+                Message_ContinueTextbox(
+                    play, sBombersNotebookEventMessages
+                              [msgCtx->bombersNotebookEventQueue[msgCtx->bombersNotebookEventQueueCount]]);
+                Audio_PlaySfx(NA_SE_SY_SCHEDULE_WRITE);
+                return true;
+            }
+        }
+    }
+}
+
+u16 D_801D028C[] = {
+    0x1B5A, 0x1B5B, 0x1B5C, 0x1B5D, 0x1B5E, 0x1B5F, 0x1B60, 0x1B61, 0x1B62, 0x1B63, 0x1B64, 0x1B65, 0x1B66,
+    0x1B67, 0x1B68, 0x1B69, 0x1B6A, 0,      0xDC7,  0xDC9,  0x120A, 0x120B, 0x122C, 0x122F, 0x120A, 0x120B,
+};
```

Comment:
Kind of a pain, but I think this is how it's mapped?
```c
u16 D_801D028C[] = {
    0x1B5A, // OCARINA_ACTION_0
    0x1B5B, // 
    0x1B5C, // OCARINA_ACTION_DEMONSTRATE_SONATA
    0x1B5D, // OCARINA_ACTION_DEMONSTRATE_GORON_LULLABY
    0x1B5E, // OCARINA_ACTION_DEMONSTRATE_NEW_WAVE
    0x1B5F, // OCARINA_ACTION_DEMONSTRATE_ELEGY
    0x1B60, // OCARINA_ACTION_DEMONSTRATE_OATH
    0x1B61, // OCARINA_ACTION_DEMONSTRATE_SARIAS
    0x1B62, // OCARINA_ACTION_DEMONSTRATE_TIME
    0x1B63, // OCARINA_ACTION_DEMONSTRATE_HEALING
    0x1B64, // OCARINA_ACTION_DEMONSTRATE_EPONAS
    0x1B65, // OCARINA_ACTION_DEMONSTRATE_SOARING
    0x1B66, // OCARINA_ACTION_DEMONSTRATE_STORMS
    0x1B67, // OCARINA_ACTION_DEMONSTRATE_SUNS
    0x1B68, // OCARINA_ACTION_DEMONSTRATE_INVERTED_TIME
    0x1B69, // OCARINA_ACTION_DEMONSTRATE_DOUBLE_TIME
    0x1B6A, // OCARINA_ACTION_DEMONSTRATE_GORON_LULLABY_INTRO
    0,      // OCARINA_ACTION_11 
    0xDC7,  // OCARINA_ACTION_3B 
    0xDC9,  // OCARINA_ACTION_3C 
    0x120A, // OCARINA_ACTION_DEMONSTRATE_EVAN_PART1_FIRST_HALF 
    0x120B, // OCARINA_ACTION_DEMONSTRATE_EVAN_PART2_FIRST_HALF 
    0x122C, // OCARINA_ACTION_DEMONSTRATE_EVAN_PART1_SECOND_HALF 
    0x122F, // OCARINA_ACTION_DEMONSTRATE_EVAN_PART2_SECOND_HALF 
    0x120A, // OCARINA_ACTION_PROMPT_EVAN_PART1_SECOND_HALF 
    0x120B, // OCARINA_ACTION_PROMPT_EVAN_PART2_SECOND_HALF
};
```
Looking at the textIds seems to indicate this is right